common/types/head.go

```diff
@@ -1,28 +1,26 @@
 package types
 
-import "github.com/ethereum/go-ethereum/common"
-
 // Head provides access to a chain's head, as needed by the TxManager.
 // This is a generic interface which ALL chains will implement.
 //
 //go:generate mockery --quiet --name Head --output ./mocks/ --case=underscore
-type Head interface {
+type Head[HASH Hashable] interface {
```

Comment:
HASH should instead be BLOCK_HASH

Reply:
Thanks for the comments Prashant! Is there a reason why it should be BLOCK_HASH instead? Is it because of naming conventions or functional reasons?